enhancements/vdpa/virtio-vdpa-ovn-hw-offload.md

```diff
@@ -0,0 +1,237 @@
+  - Install vendor-dependent kernel drivers (e.g. mlx5-core for Mellanox cards)
+  - Bind the right PCI driver (vendor specific). Some vendors might implement vdpa in a different PCI driver (e.g: Intel’s ifcvf). Others might keep the same pci driver and require extra steps (e.g: in mellanox vdpa, the VF is still bound to “mlx5_core”)
+- Create the vdpa device (vendor specific). Some vendors might require no extra steps because they create the vdpa device on a pci driver probe (e.g: ifcvf). Others might need extra steps (e.g: Mellanox requires loading an additional driver and in the future, they 	might require managing a “virtual bus”(source)). The plan is to extend the govdpa library to provide this functionality.
+- Bind the vdpa device to the correct vdpa bus driver (virtio-vdpa driver in the first implementation). This is not vendor specific and uses a sysfs-based API
+- Configure the SRIOV device plugin
+  - Add the appropriate vdpa flags to the SR-IOV Device Plugin’s configMap.
+  - Create the Network Attachment Definitions
+
+Note: the SRIOV operator (config-daemon) needs to destroy the vdpa device when policy is removed
+
+
+### API Extensions
+
+The plan is to extend the SriovNetworkNodePolicy CRD API to support the vDPA feature.
+
+There are two proposals:
+
+#### **Proposal A**
+Change the semantics of the deviceType field to express the way the device is exposed to the user and let the SR-IOV Network Operator in cooperation with the vendor plugins determine what is the driver that must be bound at the PCI bus (and, of course the vdpa bus).
+**deviceType** would assume one of the following values: netdev/vfio-pci/vdpa-virtio/vdpa-vhost.
+The problem with this approach:
+**isRdma** goes in another direction. isRdma changes the way the device is exposed to the user but is only compatible with netdev type (and only one vendor implements it AFAIK).
+So for this proposal to be complete, deviceType would also include “rdma”. This could become a compatibility issue but backwards compatibility should be manageable.
+Benefits of this proposal: Easier for the user.
+#### **Proposal B**
+Keep deviceType meaning “driver bound to PCI device”. And add a field called **vdpaType** to select the type of vdpa device (vhost/virtio).
+We need to add a runtime check to verify that a user does not specify { “deviceType”:  “vfio-pci”, “vdpaType”: “virtio” } because that doesn’t make sense.
+Benefits:
+Easier to implement
+Problems:
+If another vendor requires another driver for vdpa in the future we would need to expand deviceType to add the new driver.
+Relies more on knowledge from the user.
+Runtime errors (user creates a combination of fields that are wrong) are more difficult to return to the user.
+
+
+### Implementation Details/Notes/Constraints
+
+The proposed implementation depends on the CX-6 OVS hardware work which is not fully ready yet in OCP 4.11.
+
+**TBD: check if any dependencies with other projects, e.g. RHEL, drivers, etc**
```

Comment:
Do we have all the dependencies sorted out? can we remove it?

Reply:
done